### PR TITLE
Setup initial jenkins build

### PR DESCRIPTION
**Pull request by brandondoran**
@pierreozoux This is a first stab at a Jenkinsfile for node-builder. This is totally untested but just wanted to start somewhere on getting up to speed with Jenkins.

The short term motivation is that docker pull thredup/node-builder is failing in CircleCi. Long term, Jenkins is probably where we're moving for CI and I'd like to learn it.

Related PR: https://github.com/thredup/tup-dev-ops/pull/176

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/groovy
+podTemplate(label: 'node-builder-jenkins-pipeline', containers: [
+  containerTemplate(name: 'jnlp', image: 'jenkinsci/jnlp-slave:2.62', args: '${computer.jnlpmac} ${computer.name}', workingDir: '/home/jenkins', resourceRequestCpu: '200m', resourceLimitCpu: '200m', resourceRequestMemory: '256Mi', resourceLimitMemory: '256Mi'),
+  containerTemplate(name: 'docker', image: 'docker:1.12.6',         command: 'cat', ttyEnabled: true),
+  containerTemplate(name: 'awscli', image: 'quay.io/coreos/awscli', command: 'cat', ttyEnabled: true)
```
Comment (pierreozoux) [inline Jenkinsfile]: I think we can get rid of this container, it is not used I guess.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+              println "Promote Docker image to ECR as: 'latest'"
+              img.push("latest")
+            }
+          }
+        }
+
+        stage ('Build HUB container') {
+          println "Running Docker build for HUB:"
+          def img = docker.image hub_image
+          sh (script: "docker build -t ${hub_image} .", returnStdout: true)
```
Comment (pierreozoux) [inline Jenkinsfile]: we don't need to build 2 times? We could just tag?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          docker.withRegistry("https://index.docker.io/v1/", "docker-hub") {
```
Comment (pierreozoux) [inline Jenkinsfile]: we'll need to authenticate to the docker hub.

What do you think about using ecr registry, also for dev setup?
Then we would have one place to manage access for devs: AWS :)
What would be the benefit of using docker hub?

File: Jenkinsfile (hunk 2 above)
Comment (brandondoran) [inline Jenkinsfile]: Good point

File: Jenkinsfile (hunk 3 above)
Comment (brandondoran) [inline Jenkinsfile]: Yeah, I'm ready to move to ECR. There's no real reason we use docker hub other than it's just what we've always done in the past.

**Pull request merged**